src/posting/importing/postman.py

```diff
@@ -0,0 +1,180 @@
+from pathlib import Path
+from typing import Any, List, Optional
+import json
+import os
+import re
+import yaml
+
+from pydantic import BaseModel
+
+from rich.console import Console
+
+from posting.collection import APIInfo, Collection
+
+
+class Variable(BaseModel):
+    key: str
+    value: Optional[str] = None
+    src: Optional[str | List[str]] = None
+    fileNotInWorkingDirectoryWarning: Optional[str] = None
+    filesNotInWorkingDirectory: Optional[List[str]] = None
+    type: Optional[str] = None
+    disabled: Optional[bool] = None
+
+
+class Body(BaseModel):
+    mode: str
+    options: Optional[dict] = None
+    raw: Optional[str] = None
+    formdata: Optional[List[Variable]] = None
+
+
+class Url(BaseModel):
+    raw: str
+    host: Optional[List[str]] = None
+    path: Optional[List[str]] = None
+    query: Optional[List[Variable]] = None
+
+
+class PostmanRequest(BaseModel):
+    method: str
+    url: Optional[str | Url] = None
+    header: Optional[List[Variable]] = None
+    description: Optional[str] = None
+    body: Optional[Body] = None
+
+
+class RequestItem(BaseModel):
+    name: str
+    item: Optional[List["RequestItem"]] = None
+    request: Optional[PostmanRequest] = None
+
+
+class PostmanCollection(BaseModel):
+    info: dict[str, str]
+    item: List[RequestItem]
+    variable: List[Variable]
+
+
+def create_env_file(path: Path, env_filename: str, variables: List[Variable]) -> Path:
+    env_content: List[str] = []
+
+    for var in variables:
+        env_content.append(f"{transform_variables(var.key)}={var.value}")
+
+    env_file = path / env_filename
+    env_file.write_text("\n".join(env_content))
+    return env_file
+
+
+def generate_directory_structure(
+    items: List[RequestItem], current_path: str = "", base_path: Path = Path("")
+) -> List[str]:
+    directories = []
+    for item in items:
+        if item.item is not None:
+            folder_name = item.name
+            new_path = f"{current_path}/{folder_name}" if current_path else folder_name
+            full_path = Path(base_path) / new_path
+            os.makedirs(str(full_path), exist_ok=True)
+            directories.append(str(full_path))
+            generate_directory_structure(item.item, new_path, base_path)
+        if item.request is not None:
+            request_name = re.sub(r"[^A-Za-z0-9\.]+", "", item.name)
+            file_name = f"{request_name}.posting.yaml"
+            full_path = Path(base_path) / current_path / file_name
+            create_request_file(full_path, item)
+    return directories
```

Comment:
Ideally we'd build a `Collection` just like the "import from OpenAPI" feature, then just call `save_to_disk`.

That would also tidy up the `import_spec` function, as it would mean that handling would be consistent across OpenAPI and Postman.

Reply:
Ah, I see! Let me update that